### ISSUE TITLE
Haxelib availability on Mac using Crossover

### ISSUE THREAD

**Issue by larsiusprime** [fdorg/flashdevelop#818]
I've been using FD on Mac using Crossover and it actually works pretty great!

The only main issue I have is that I can't get it to detect my haxelibs. I followed the tutorial video, installed via Crossover, and set up the FlashDevelop Bridge thingy as instructed. However I have no clue how to properly set up Haxe/Haxelib.

Some relevant infos:

* Haxe is installed on the mac, the normal way.
* All my haxe libraries are installed on the mac, the normal way.
* I used AppMan in FD to download and install Haxe/Neko 3.2.0 (presumably inside crossover/wine?)

I don't need to be able to compile from FlashDevelop, all I really need is to be able to resolve the haxelibs well enough to be able to get at all my files from one place. If I have to, I can just compile from the command line. I just like using FD as my editor environment.

**Comment by Meychi** [comment]
Haxe works with FD pretty well in CrossOver. I have been using it like this on my Mac. In your setup you use the CrossOver installed Haxe and not the Mac's. Easiest way is to use CrossOver Haxe fully.

Can you set custom Haxe lib directory somehow? If so, you could map your libs to a Windows drive and then make CrossOver Haxe to use the Mac libs instead.

**Comment by larsiusprime** [comment]
Okay so as I understand it:

1) Install Haxe toolkit (compiler, standard library, neko, and haxelib utility) via Crossover? (ie in windows land)
2) Put my haxe library files in a shared location (ie in windows land)
3) Tell FD to use the windows-land version of haxe?

Then back in mac-land:

4) Install Haxe toolkit on the mac
5) Use "haxelib dev" command to point my mac version of haxelib to the libraries on the shared location

And then I can use FD on mac, but also compile native mac binaries from the mac too?

**Comment by Meychi** [comment]
Yes. Or otherway around. I have used the same Windows Flex+AIR SDK on Mac and CrossOver. It should work but i need to test it. Compilation may not work if there are binaries in the libs.

**Comment by elsassph** [comment]
@Meychi somehow I think it would be better to keep the haxe compiler and libs on Mac land, and reinstall haxe compiler on Windows land and link to your Mac's haxelibs using a share point. I think this would be the less friction and wasted disk space, especially if you need a C++ toolchain for your project.

**Comment by Meychi** [comment]
That is why i said or otherway around. :) Ok, got this fully working with minimal effort. Libs found and FD even compiles a Haxe JS project using Actuate lib. Here are the steps:

1. Install Haxe from AppMan for CrossOver
2. Open CrossOver command prompt from FD toolbar
3. Run haxelib setup
4. Give it your Mac Haxe lib path in Windows format. Mine was: Z:\usr\lib\haxe\lib
5. Ready to rock with you Mac Haxe libs :)

**Comment by larsiusprime** [comment]
Neat! I'll try it out soon as I can and confirm whether it works for me.

**Comment by larsiusprime** [comment]
Okay, thanks guys. It works!

Some notes:

The method you've described works great for haxe libraries you install the regular way, but it has a glitch with anything you install via "haxelib dev." Since you're pointing both windows-land and mac-land haxelib utility to the same path, it's using the exact same metadata to define the library environment for both native mac and wine, but they ask for different path formats.

So to get things to work in FD, you have to open a wine cmd window and type ```haxelib foo Z:/path/to/foo``` but then if you want to compile natively on mac, you have to change that back to ```haxelib foo path/to/foo```.

So I wonder if there's just a way to be slightly more strategic about how you set up haxelib to get this working. For instance, you could just revert haxelib setup in wine to its default location and kind of laboriously add your haxelibs via ``haxelib dev abc Z:/path/to/abc/on/the/mac``` for each library.

**Comment by Meychi** [comment]
I use them the regular way :) I knew this path issue might be a problem in some cases.

Maybe you could create a bat script for CrossOver/Wine that reads the lib directories from Mac lib dir and and updates the dev paths in CrossOver/Wine Haxe for you when ran.

**Comment by elsassph** [comment]
Ah snap, dev paths will be quite a problem... Sounds like the original idea of defining everything as haxelib dev paths is the way to go.

**Comment by larsiusprime** [comment]
Honestly if haxelib wasn't a piece of junk and made it easier to hot-swap an entire set of library configurations per project, then it would be super easy to just make use of that factor here. Hmm, let me think what I can do to make this easier...
